script/CheckSecurityConfigs.s.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { console2 } from "forge-std/console2.sol";
+import { Script } from "forge-std/Script.sol";
+import { stdStorage } from "forge-std/StdStorage.sol";
+
+/**
+ * @title CheckSecuityConfigs
+ * @notice A script to check security configurations of an OP Chain,
+           such as upgrade key holder, challenger and guadian designations.
+ *         The usage is as follows:
+ *         $ forge script CheckSecuityConfigs \
+ *             --rpc-url $ETH_RPC_URL
+ */
+
+contract CheckSecuityConfigs is Script {
+    struct ProtocolAddresses {
+        // Protocol contracts
+        address AddressManager;
+        address L1CrossDomainMessengerProxy;
+        address L1ERC721BridgeProxy;
+        address L1StandardBridgeProxy;
+        address L2OutputOracleProxy;
+        address OptimismMintableERC20FactoryProxy;
+        address OptimismPortalProxy;
+        address ProxyAdmin;
+        address SystemConfigProxy;
+
+        // Privileged roles
+        address Challenger;
+        address Guardian;
+        address ProxyAdminOwner;
+        address SystemConfigOwner;
+    }
+
+    bool internal hasErrors;
+
+    /**
+     * @notice The entrypoint function.
+     */
+    function run() external {
+        string[4] memory addressesJsonFiles = [
+            "superchain/extra/addresses/mainnet/base.json",
+            "superchain/extra/addresses/mainnet/op.json",
+            "superchain/extra/addresses/mainnet/pgn.json",
+            "superchain/extra/addresses/mainnet/zora.json"
+        ];
+        hasErrors = false;
+        for(uint i = 0; i < addressesJsonFiles.length; i++) {
+            runOnSingleFile(addressesJsonFiles[i]);
+        }
+        assert(!hasErrors);
+    }
+
+    function runOnSingleFile(string memory addressesJsonPath) internal {
+        console2.log("Checking %s", addressesJsonPath);
+        ProtocolAddresses memory addresses = getAddresses(addressesJsonPath);
+        checkAddressManager(addresses);
+        checkL1CrossDomainMessengerProxy(addresses);
+        checkL1ERC721BridgeProxy(addresses);
+        checkL1StandardBridgeProxy(addresses);
+        checkL2OutputOracleProxy(addresses);
+        checkOptimismMintableERC20FactoryProxy(addresses);
+        checkOptimismPortalProxy(addresses);
+        checkProxyAdmin(addresses);
+        checkSystemConfigProxy(addresses);
+        // TODO: Check implementations.
```

Comment:
Can you open a github issue for this, and expand the description there on what needs to be checked?